src/xds.c

```diff
@@ -25,38 +25,154 @@
 #include "xds.h"
 #include "caption.h"
 #include <string.h>
+#include <xds.h>
+#include <caption.h>
+#include <xds_data.h>
+#include <stdio.h>
```

Comment:
nitpicking, but should probably remove stdio header if it is not used

Reply:
it's a valid criticism, I forgot to do so after testing

Reply:
Are you going to remove it?